ui/app/routes/optimization/supervised-fine-tuning/model_options.ts

```diff
@@ -8,6 +8,42 @@ export const ModelOptionSchema = z.object({
 
 export type ModelOption = z.infer<typeof ModelOptionSchema>;
 
+/**
+ * Detect provider from model name prefix for autocomplete.
+ * Best-effort matching for common model naming patterns.
+ */
+export function detectProviderFromModelName(
+  modelName: string,
+): ModelOption["provider"] {
+  const lower = modelName.toLowerCase();
+
+  // GCP Vertex Gemini
+  if (lower.startsWith("gemini")) {
+    return "gcp_vertex_gemini";
+  }
+
+  // Together AI
+  if (
+    lower.startsWith("gemma") ||
+    lower.startsWith("qwen") ||
+    lower.startsWith("deepseek")
+  ) {
+    return "together";
+  }
+
+  // Fireworks
+  if (
+    lower.startsWith("llama") ||
+    lower.startsWith("mixtral") ||
+    lower.startsWith("mistral") ||
+    lower.startsWith("yi")
+  ) {
+    return "fireworks";
+  }
+
+  return "openai";
+}
```

Comment:
I think we can skip this function completely. Together and Fireworks offer roughly the same models (we just put different ones as an example), so it probably doesn't make sense to prefill either here. And over time it'll get harder to maintain.